alt_e2eshark/e2e_testing/backends.py

```diff
@@ -96,12 +101,15 @@ def __init__(self, *, device="local-task", hal_target_backend="llvm-cpu", target
                 "--iree-llvmcpu-target-cpu=host",
             ]
     
-    def compile(self, module_path: str, *, save_to : str = None) -> str:
+    def compile(self, module_path: str, *, save_to : str = None, extra_options : CompilerOptions) -> str:
+        test_specific_args = extra_options.common_extra_args
+        if self.hal_target_backend in extra_options.backend_specific_flags.keys():
+            test_specific_args += extra_options.backend_specific_flags[self.hal_target_backend]
+        compile_args = self.extra_args + [flag(arg) for arg in test_specific_args]
         vmfb_path = os.path.join(save_to, "compiled_model.vmfb")
         arg_string = f"--iree-hal-target-backends={self.hal_target_backend} "
-        for arg in self.extra_args:
-            arg_string += arg
-            arg_string += " "
+        for arg in compile_args:
+            arg_string += f'{arg} '
```

Comment:
I know a neat one-liner to replace the for loop :)
```suggestion
            arg_string += " ".join([*compile_args, ""])
```
The empty string should append a whitespace to `arg_string`.